### PR TITLE
demo for 2D slice analysis based on 3D inputs

### LINKED ISSUE #130: tutorial/demo for 2D slice analysis based on 3D inputs
**Is your feature request related to a problem? Please describe.**
would be great to demonstrate the available options for both training and inference

there were quite a few requests, see also:
- https://github.com/Project-MONAI/MONAI/discussions/1641
- https://github.com/Project-MONAI/MONAI/issues/719
- https://github.com/Project-MONAI/MONAI/issues/995
- https://github.com/Project-MONAI/MONAI/issues/808

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #130
- basic binary seg example with random 2D samples + 2d unet for 3D inputs
- inference with a sliceinferer

Screenshot 2022-06-05 at 22 07 12

### Status
**Ready**

### Checks

- [ ] Notebook runs automatically `./runner [-p <regex_pattern>]`

### CODE REVIEW

File: modules/training_with_2d_slices.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    # -----
+    # patch-level preprocessing
+    # -----
+    # resize the sampled slices to a consistent size so that we can batch
+    # the last spatial dim is always 1, so we squeeze dim.
+    patch_transform = Compose(
+        [
+            SqueezeDimd(keys=["img", "seg"], dim=-1),  # squeeze the last dim
+            Resized(keys=["img", "seg"], spatial_size=[48, 48]),
+            # ResizeWithPadOrCropd(keys=["img", "seg"], spatial_size=[48, 48], mode="replicate"),
```
Comment (Nic-Ma) [inline modules/training_with_2d_slices.py:111]: please remove the commented code.

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Hi @wyli ,

Thanks for the quick demo.
Could you please help add more description for the 2 typical methods in this example?
(1) Randomly crop slices for training, (2) Slices for inference.
Or maybe it's better to convert it into a jupyter notebook? So we can include the plot charts of `matshow3d` or image slices in it?

Thanks.

**Comment by wyli** [comment]
thanks, I'll try to convert this scripti into a notebook example.

**Comment by wyli** [comment]
merging these notebooks @Nic-Ma @dannyhow12, please let me know any further questions.

**Pull request merged**